### PR TITLE
CNV-38571: Monitor user interaction in InstanceTypes and Templates flow

### PR DESCRIPTION
**Pull request by pcbailey**
## 📝 Description

This PR adds monitoring for events in the InstanceType and Template VM creation flows.

Jira: https://issues.redhat.com/browse/CNV-38571

### CODE REVIEW

File: src/utils/extensions/telemetry/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { kubevirtConsole as kvConsole } from '@kubevirt-utils/utils/utils';
```
Comment (metalice) [inline src/utils/extensions/telemetry/telemetry.ts:1]: why changing its name?

File: src/utils/extensions/telemetry/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+const SEGMENT_KEY =
```
Comment (metalice) [inline src/utils/extensions/telemetry/telemetry.ts:3]: Am I guessing all these files are from the Console? or some kind of demo?

File: src/utils/extensions/telemetry/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      analytics.load(SEGMENT_KEY);
+    }
+  }
+};
+
+initSegment();
+
+export const eventMonitor = async (eventType: string, properties?: any) => {
+  // eslint-disable-next-line no-console
+  console.log('Demo Plugin received telemetry event: ', eventType, properties);
```
Comment (metalice) [inline src/utils/extensions/telemetry/telemetry.ts:76]: can be removed

File: src/views/catalog/CustomizeInstanceType/components/CustomizeITVMFooter.tsx [hunk 4]
```diff
@@ -75,6 +82,10 @@ const CustomizeITVMFooter: FC = () => {
                       data: vmSignal.value || vm,
                       model: VirtualMachineModel,
                     });
+                    eventMonitor(CUSTOMIZE_VM_SUCCEEDED, {
```
Comment (metalice) [inline src/views/catalog/CustomizeInstanceType/components/CustomizeITVMFooter.tsx:85]: i can see in the rest of the PR that u always send a vmName and most of the time a flow, why not creating 3 funcs,

eventMonitorWithName , eventMonitorITFlow , eventMonitorTemplateFlow , the 2 last will use eventMonitorWithName, and u simply sending the VM and the key, all the rest of logic of adding a flow and a name will be under those 3 funcs, u wont need to import getName on each file, and simpler to work with

File: src/utils/extensions/telemetry/telemetry.ts (hunk 1 above)
Comment (pcbailey) [reply src/utils/extensions/telemetry/telemetry.ts]: I don't remember why I did that. I changed it back.

File: src/utils/extensions/telemetry/telemetry.ts (hunk 2 above)
Comment (pcbailey) [reply src/utils/extensions/telemetry/telemetry.ts:7]: Yes, this is how the API key is retrieved in Console.

File: src/utils/extensions/telemetry/telemetry.ts (hunk 3 above)
Comment (pcbailey) [reply src/utils/extensions/telemetry/telemetry.ts]: Done!

File: src/views/catalog/CustomizeInstanceType/components/CustomizeITVMFooter.tsx (hunk 4 above)
Comment (pcbailey) [reply src/views/catalog/CustomizeInstanceType/components/CustomizeITVMFooter.tsx]: Done! There were some places where the VM wasn't accessible, so I had to work around those.

**Pull request merged**